tests/unit/d.ts

```diff
@@ -57,24 +67,42 @@ registerSuite({
 			assert.isTrue(isWNode(dNode));
 			assert.isFalse(isHNode(dNode));
 		},
-		'create WNode wrapper using constructor with properties and children'() {
-			const properties = { mand: true };
-			const dNode = w(TestWidget, properties, [ w(WidgetBase, properties) ]);
+		'create WNode wrapper using constructor with children'() {
+			const dNode = w(TestWidget, { required: true }, [ w(TestChildWidget, { myChildProperty: '' }) ]);
 
 			assert.equal(dNode.type, WNODE);
 			assert.deepEqual(dNode.widgetConstructor, TestWidget);
-			assert.deepEqual(dNode.properties, { mand: true });
+			assert.lengthOf(dNode.children, 1);
+			assert.isTrue(isWNode(dNode));
+			assert.isFalse(isHNode(dNode));
+		},
+		'create WNode wrapper using constructor with VNode children'() {
+			const dNode = w(TestChildWidget, { myChildProperty: '' }, [ v('div') ]);
+
+			assert.equal(dNode.type, WNODE);
+			assert.deepEqual(dNode.widgetConstructor, TestChildWidget);
+			assert.lengthOf(dNode.children, 1);
+			assert.isTrue(isWNode(dNode));
+			assert.isFalse(isHNode(dNode));
+		},
+		'create WNode wrapper using constructor with properties and children'() {
+			const properties: any = { id: 'id', classes: [ 'world' ] };
+			const dNode = w(WidgetBase, properties, [ w(WidgetBase, properties) ]);
+
+			assert.equal(dNode.type, WNODE);
+			assert.deepEqual(dNode.widgetConstructor, WidgetBase);
+			assert.deepEqual(dNode.properties, { id: 'id', classes: [ 'world' ]});
 			assert.lengthOf(dNode.children, 1);
 			assert.isTrue(isWNode(dNode));
 			assert.isFalse(isHNode(dNode));
 		},
 		'create WNode wrapper using label with properties and children'() {
-			const properties = { mand: false };
-			const dNode = w<TestProperties>('my-widget', properties, [ w(WidgetBase, properties) ]);
+			const properties: any = { id: 'id', classes: [ 'world' ] };
```

Comment:
I assume `w()` no longer likes the inferred type?  Is this going to become frustrating/challenging, or do you think this won't cause too much `any` ugliness in real life?